### DISCUSSION TITLE
What is the difference between wp.dot() and dot()

### DISCUSSION THREAD

**Discussion by cadop** [NVIDIA/warp#6] [Q&A] [answered]
In a few places there are calls to a math operation like `dot`, and in the same function there is `wp.dot` (same for length).  
Are these used for different things, or what goes in to deciding when to use which one?

![image](https://user-images.githubusercontent.com/11399119/159734602-fd26ff5f-3042-4b4e-9649-4feb9f94d8f8.png)

**Comment by mmacklin** [comment, marked as answer]
Hi @cadop , there is no difference - currently the `wp` namespace can be considered 'implicit' inside kernel code for all built-in functions, so using the wp. prefix is optional (but generally preferred).
